### PR TITLE
NAS-124320 / 24.04 / Added exporters forms and tables

### PR DESCRIPTION
**Pull request by RehanY147**
Go to Reporting -> Exporters (top right)
Check the table and the add/edit form. Also test the enable/disable toggle button on the table.

### CODE REVIEW

**Review by dszidi — COMMENTED** [review]
Checked on the team's M40 as well as on my own VM. I'm getting errors. Does this require a particular middleware build? 

- Table has warning that it "Can not retrieve response"
- On the form, the field "Reporting Exporter Type" is disabled
- The loading animation on the form is infinite
- Unable to Save form

image

**Comment by RehanY147** [comment]
@dszidi m40 is on Cobia. These changes are for Dragonfish

File: src/app/pages/reports-dashboard/components/exporters/reporting-exporters-form/reporting-exporters-form.component.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      (submit)="onSubmit()"
+    >
+      <ix-fieldset>
+        <ix-input
+          formControlName="name"
+          [label]="'Name' | translate"
+          [tooltip]="'Give your exporter configuration a name'"
+          [required]="true"
+        ></ix-input>
+        <ix-select
```
Comment (undsoft) [inline src/app/pages/reports-dashboard/components/exporters/reporting-exporters-form/reporting-exporters-form.component.html:17]: [required]="true"

File: src/app/pages/reports-dashboard/components/exporters/reporting-exporters-form/reporting-exporters-form.component.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          formControlName="type"
+          [label]="'Reporting Exporter Type' | translate"
+          [options]="exporterTypeOptions$"
+          (ngModelChange)="onExporterTypeChanged($event)"
```
Comment (undsoft) [inline src/app/pages/reports-dashboard/components/exporters/reporting-exporters-form/reporting-exporters-form.component.html:21]: It's better to use valueChanges with reactive forms. This is kinda non-standard.

File: src/app/pages/reports-dashboard/components/exporters/reporting-exporters-form/reporting-exporters-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+  get title(): string {
+    return this.isNew
+      ? this.translate.instant('Add Reporting Exporter')
+      : this.translate.instant('Edit Reporting Exporter');
+  }
+
+  form = this.fb.group({
+    name: [null as string, Validators.required],
+    enabled: [false as boolean],
```
Comment (undsoft) [inline src/app/pages/reports-dashboard/components/exporters/reporting-exporters-form/reporting-exporters-form.component.ts:37]: I think we can default to true. Cast is unnecessary.

File: src/app/pages/reports-dashboard/components/exporters/reporting-expoters-list/reporting-exporters-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      error: () => {
+        this.dataProvider.setRows([]);
+        this.isLoading$.next(false);
+        this.hasError$.next(true);
+        this.cdr.markForCheck();
+      },
+    });
+  }
+
+  onListFiltered(query: string): void {
```
Comment (undsoft) [inline src/app/pages/reports-dashboard/components/exporters/reporting-expoters-list/reporting-exporters-list.component.ts:120]: This is not connected to anything.

File: src/app/pages/reports-dashboard/components/exporters/reporting-expoters-list/reporting-exporters-list.component.html [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+              [attr.aria-label]="'Delete' | translate"
+              [matTooltip]="'Delete' | translate"
+              [ixTest]="['delete', exporter.id]"
+              (click)="doDelete(exporter)"
+            ><ix-icon name="mdi-delete"></ix-icon></button>
+          </div>
+        </ng-template>
+      </tbody>
+    </ix-table2>
+    <ix-table-pager-show-more [pageSize]="10" [dataProvider]="dataProvider"></ix-table-pager-show-more>
```
Comment (undsoft) [inline src/app/pages/reports-dashboard/components/exporters/reporting-expoters-list/reporting-exporters-list.component.html:51]: I think it's best to use normal pager for full-width tables like these.

**Review by undsoft — APPROVED** [review]
Looks good. Can we also check with middleware to see if they can have better titles in schema, so that it's not `destination_ip`, but `Destination IP`?

**Comment by RehanY147** [comment]
backport

**Pull request merged**